src/source/source_cache.js

```diff
@@ -490,52 +495,48 @@ class SourceCache extends Evented {
         // parent or child tiles that are *already* loaded.
         const retain = this._updateRetainedTiles(idealTileIDs, zoom);
 
-        const parentsForFading = {};
-
         if (isRasterType(this._source.type)) {
+            const parentsForFading = {};
+            const fadingTiles = {};
             const ids = Object.keys(retain);
-            for (let k = 0; k < ids.length; k++) {
-                const id = ids[k];
+            for (const id of ids) {
                 const tileID = retain[id];
                 assert(tileID.key === +id);
+
                 const tile = this._tiles[id];
-                if (!tile) continue;
-
-                // If the drawRasterTile has never seen this tile, then
-                // tile.fadeEndTime may be unset.  In that case, or if
-                // fadeEndTime is in the future, then this tile is still
-                // fading in. Find tiles to cross-fade with it.
```

Comment:
I felt this would be redundant due to the comment after that. So one would say "skip tiles that not loaded or already faded in", and another "do X with tiles that are loaded but still fading in".